Index: src/ui/VirtualList.java
===================================================================
--- src/ui/VirtualList.java	(revision 478)
+++ src/ui/VirtualList.java	(working copy)
@@ -86,6 +86,10 @@
     public static final int NOKIA_GREEN=-10;
     public static final int MOTOROLA_GREEN=-10;
     public static final int SE_GREEN=0;
+    public static final int MOTOE680_VOL_UP=-9;
+    public static final int MOTOE680_VOL_DOWN=-8;
+    public static final int MOTOE680_REALPLAYER=-6;
+    public static final int MOTOE680_FMRADIO=-7;
     
     public static int greenKeyCode=SIEMENS_GREEN;
     public static boolean fullscreen=false;
@@ -549,7 +553,9 @@
     private void key(int keyCode) {
         switch (keyCode) {
             case 0: break;
+            case MOTOE680_VOL_UP:
             case KEY_NUM1:  { moveCursorHome();    break; }
+            case MOTOE680_VOL_DOWN:
             case KEY_NUM7:  { moveCursorEnd();     break; }
             case '5':{ eventOk(); break; }
             default:
Index: src/Client/Roster.java
===================================================================
--- src/Client/Roster.java	(revision 478)
+++ src/Client/Roster.java	(working copy)
@@ -1122,7 +1122,7 @@
     }
     
     public void userKeyPressed(int keyCode){
-        if (keyCode==KEY_NUM0) {
+        if (keyCode==KEY_NUM0 || keyCode==MOTOE680_REALPLAYER) {
             if (messageCount==0) return;
             Object atcursor=getFocusedObject();
             Contact c=null;
@@ -1225,14 +1225,14 @@
         if (keyCode==cf.keyLock) 
             new KeyBlock(display, getTitleItem(), cf.keyLock, cf.ghostMotor); 
 
-        if (keyCode==cf.keyVibra) {
+        if (keyCode==cf.keyVibra || keyCode==MOTOE680_FMRADIO) {
             cf.profile=(cf.profile==AlertProfile.VIBRA)? 
                 cf.def_profile : AlertProfile.VIBRA;
             updateTitle();
             redraw();
         }
         
-        if (keyCode==cf.keyOfflines) {
+        if (keyCode==cf.keyOfflines || keyCode==MOTOE680_REALPLAYER) {
             cf.showOfflineContacts=!cf.showOfflineContacts;
             reEnumRoster();
         }
