# Мэпинг клаиши RealPlayer в Motorola E680(i) для закрытия диалогов
# выбора статуса, стиля оповещения, учетной записи
# (c) Dmitry Nezhevenko <dion@inhex.net>
Index: src/Client/StatusSelect.java
===================================================================
--- src/Client/StatusSelect.java	(revision 480)
+++ src/Client/StatusSelect.java	(working copy)
@@ -145,5 +145,10 @@
         }
     }
     
+    public void userKeyPressed(int keyCode) {
+        if (keyCode==MOTOE680_REALPLAYER) {
+                destroyView();
+                return;
+        }
+    }
 }
-
Index: src/Client/AlertProfile.java
===================================================================
--- src/Client/AlertProfile.java	(revision 480)
+++ src/Client/AlertProfile.java	(working copy)
@@ -112,4 +112,11 @@
         }
         if (notify!=null) notify.startNotify();
     }
+
+    public void userKeyPressed(int keyCode) {
+        if (keyCode==MOTOE680_REALPLAYER) {
+                destroyView();
+                return;
+        }
+    }
 }
Index: src/Client/AccountSelect.java
===================================================================
--- src/Client/AccountSelect.java	(revision 480)
+++ src/Client/AccountSelect.java	(working copy)
@@ -137,5 +137,11 @@
         NvStorage.writeFileRecord(outputStream, Account.storage, 0, true);
     }
 
+    public void userKeyPressed(int keyCode) {
+        if (keyCode==MOTOE680_REALPLAYER) {
+                destroyView();
+                return;
+        }
+    }
+
 }
-
