Index: src/locale/SR.java
===================================================================
--- src/locale/SR.java	(revision 459)
+++ src/locale/SR.java	(working copy)
@@ -17,96 +17,96 @@
 //--msgstr "Jid"
     
 // #: /home/avl/Projects/bombus/src/midlet/Bombus.java:48
-    public final static String MS_LOADING = "Loading";
+    public final static String MS_LOADING = "Запуск";
 //--msgstr "Запуск"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterToolsMenu.java:31
-    public final static String MS_PRIVACY_LISTS = "Privacy Lists";
+    public final static String MS_PRIVACY_LISTS = "Списки Приватности";
 //--msgstr "Списки Приватности"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:57
-    public final static String MS_EXISTING_GROUPS = "Existing groups";
+    public final static String MS_EXISTING_GROUPS = "Существующие группы";
 //--msgstr "Существующие группы"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:158
-    public final static String MS_MESSAGE_FONT = "Message font";
+    public final static String MS_MESSAGE_FONT = "Шрифт сообщения";
 //--msgstr "Шрифт сообщения"
-    public final static String MS_ROSTER_FONT = "Roster font";
+    public final static String MS_ROSTER_FONT = "Шрифт ростера";
 //--msgstr "Шрифт ростера"
     
 // #: /home/avl/Projects/bombus/src/archive/ArchiveList.java:32
-    public final static String MS_PASTE_BODY = "Paste Body";
+    public final static String MS_PASTE_BODY = "Вставить тело сообщ.";
 //--msgstr "Вставить тело сообщ."
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:79
-    public final static String MS_CONFIG_ROOM = "Configure Room";
+    public final static String MS_CONFIG_ROOM = "Конфигуратор Комнаты";
 //--msgstr "Конфигуратор Комнаты"
     
 // #: /home/avl/Projects/bombus/src/archive/ArchiveList.java:33
-    public final static String MS_PASTE_SUBJECT = "Paste Subject";
+    public final static String MS_PASTE_SUBJECT = "Вставить тему сообщ.";
 //--msgstr "Вставить тему сообщ."
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:85
 // #: /home/avl/Projects/bombus/src/Client/RosterToolsMenu.java:30
-    public final static String MS_DISCO = "Service Discovery";
+    public final static String MS_DISCO = "Обзор Сервисов";
 //--msgstr "Обзор Сервисов"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:51
-    public final static String MS_USER_JID = "User JID";
+    public final static String MS_USER_JID = "JID пользователя";
 //--msgstr "JID пользователя"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacySelect.java:37
-    public final static String MS_NEW_LIST = "New list";
+    public final static String MS_NEW_LIST = "Новый список";
 //--msgstr "Новый список"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountSelect.java:31
-    public final static String MS_NOLOGIN = "Select (no login)";
+    public final static String MS_NOLOGIN = "Выбрать (без логина)";
 //--msgstr "Выбрать (без логина)"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyForm.java:31
-    public final static String MS_PRIVACY_RULE = "Privacy rule";
+    public final static String MS_PRIVACY_RULE = "Правило приватности";
 //--msgstr "Правило приватности"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:58
-    public final static String MS_SSL = "use SSL";
+    public final static String MS_SSL = "использовать SSL";
 //--msgstr "использовать SSL"
     
 // #: /home/avl/Projects/bombus/src/Conference/affiliation/Affiliations.java:43
-    public final static String MS_MODIFY = "Modify";
+    public final static String MS_MODIFY = "Изменить";
 //--msgstr "Изменить"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:93
-    public final static String MS_UPDATE = "Update";
+    public final static String MS_UPDATE = "Обновить";
 //--msgstr "Обновить"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:67
-    public final static String MS_ACCOUNT_NAME = "Account name";
+    public final static String MS_ACCOUNT_NAME = "Имя аккаунта";
 //--msgstr "Имя аккаунта"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:144
-    public final static String MS_GMT_OFFSET = "GMT offset";
+    public final static String MS_GMT_OFFSET = "Часовой пояс (GMT+)";
 //--msgstr "Часовой пояс (GMT+)"
     
-    public final static String MS_TIME_SETTINGS = "Time settings (hours)\n";
+    public final static String MS_TIME_SETTINGS = "Настройки времени\n";
 //--msgstr"Настройки времени\n"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:704
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:724
-    public final static String MS_CONNECTED = "Connected";
+    public final static String MS_CONNECTED = "Соединение установлено";
 //--msgstr "Соединение установлено"
     
-    public final static String MS_CONNECT_TO = "Connect to ";
+    public final static String MS_CONNECT_TO = "Соединение с";
 //--msgstr "Соединение с"
 // #: /home/avl/Projects/bombus/src/Client/AlertProfile.java:44
-    public final static String MS_ALERT_PROFILE = "Alert Profile";
+    public final static String MS_ALERT_PROFILE = "Сигналы";
 //--msgstr "Сигнализация"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyModifyList.java:36
-    public final static String MS_MOVE_UP = "Move Up";
+    public final static String MS_MOVE_UP = "Сместить вверх";
 //--msgstr "Передв. вверх"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:80
-    public final static String MS_OWNERS = "Owners";
+    public final static String MS_OWNERS = "Владельцы";
 //--msgstr "Владельцы"
     
 // #: /home/avl/Projects/bombus/src/ui/Menu.java:23
@@ -115,149 +115,149 @@
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:70
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:32
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyForm.java:40
-    public final static String MS_OK = "Ok";
+    public final static String MS_OK = "ОК";
 //--msgstr "ОК"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:96
-    public final static String MS_APP_MINIMIZE = "Minimize";
+    public final static String MS_APP_MINIMIZE = "Свернуть";
 //--msgstr "Свернуть"
     
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:49
-    public final static String MS_ROOM = "Room";
+    public final static String MS_ROOM = "Комната";
 //--msgstr "Комната"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:107
-    public final static String MS_MESSAGES = "Messages";
+    public final static String MS_MESSAGES = "Сообщения";
 //--msgstr "Сообщения"
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/ServiceDiscovery.java:43
 // #: /home/avl/Projects/bombus/src/Conference/Bookmarks.java:34
 // #: /home/avl/Projects/bombus/src/vcard/vCardForm.java:28
-    public final static String MS_REFRESH = "Refresh";
+    public final static String MS_REFRESH = "Обновить";
 //--msgstr "Обновить"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:47
-    public final static String MS_RESOLVE_NICKNAMES = "Resolve Nicknames";
+    public final static String MS_RESOLVE_NICKNAMES = "Преобразовать в ник";
 //--msgstr "Преобразовать в ник"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyForm.java:32
-    public final static String MS_PRIVACY_ACTION = "Action";
+    public final static String MS_PRIVACY_ACTION = "Действие";
 //--msgstr "Действие"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:63
-    public final static String MS_BAN = "Ban";
+    public final static String MS_BAN = "Бан";
 //--msgstr "Бан"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:81
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:77
-    public final static String MS_LEAVE_ROOM = "Leave Room";
+    public final static String MS_LEAVE_ROOM = "Покинуть Комнату";
 //--msgstr "Покинуть Комнату"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:53
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:58
-    public final static String MS_PASSWORD = "Password";
+    public final static String MS_PASSWORD = "Пароль";
 //--msgstr "Пароль"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:79
-    public final static String MS_ITEM_ACTIONS = "Actions >";
+    public final static String MS_ITEM_ACTIONS = "Действия >";
 //--msgstr "Действия >"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacySelect.java:35
-    public final static String MS_ACTIVATE = "Activate";
+    public final static String MS_ACTIVATE = "Активировать";
 //--msgstr "Активировать"
     
 // #: /home/avl/Projects/bombus/src/Conference/affiliation/AffiliationModify.java:28
-    public final static String MS_AFFILIATION = "Affiliation";
+    public final static String MS_AFFILIATION = "Членство";
 //--msgstr "Членство"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountSelect.java:43
-    public final static String MS_ACCOUNTS = "Accounts";
+    public final static String MS_ACCOUNTS = "Аккаунты";
 //--msgstr "Аккаунты"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacySelect.java:38
-    public final static String MS_DELETE_LIST = "Delete list";
+    public final static String MS_DELETE_LIST = "Удалить список";
 //--msgstr "Удалить список"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:92
-    public final static String MS_ACCOUNT_= "Account >";
+    public final static String MS_ACCOUNT_= "Аккаунт >";
 //--msgstr "Аккаунт >"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:87
-    public final static String MS_SHOWOFFLINES = "Show Offlines";
+    public final static String MS_SHOWOFFLINES = "Показать отключенных";
 //--msgstr "Показать отключенных"
     
 // #: /home/avl/Projects/bombus/src/Client/AlertProfile.java:35
 // #: /home/avl/Projects/bombus/src/Client/ActiveContacts.java:32
 // #: /home/avl/Projects/bombus/src/Client/StatusSelect.java:22
 // #: /home/avl/Projects/bombus/src/Client/SmilePicker.java:50
-    public final static String MS_SELECT = "Select";
+    public final static String MS_SELECT = "Выбрать";
 //--msgstr "Выбрать"
     
 // #: /home/avl/Projects/bombus/src/Client/MessageEdit.java:41
-    public final static String MS_SUBJECT = "Subject";
+    public final static String MS_SUBJECT = "Тема";
 //--msgstr "Тема"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:83
-    public final static String MS_GROUP_MENU = "Group menu";
+    public final static String MS_GROUP_MENU = "Меню групп";
 //--msgstr "Меню групп"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:97
 // #: /home/avl/Projects/bombus/src/Client/AccountSelect.java:36
-    public final static String MS_APP_QUIT = "Quit";
+    public final static String MS_APP_QUIT = "Выход";
 //--msgstr "Выход"
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/ServiceDiscovery.java:45
-    public final static String MS_ROSTERADD = "Add to roster";
+    public final static String MS_ROSTERADD = "Добавить в ростер";
 //--msgstr "Добавить в ростер"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacySelect.java:39
-    public final static String MS_EDIT_LIST = "Edit list";
+    public final static String MS_EDIT_LIST = "Редактировать список";
 //--msgstr "Редактировать список"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountRegister.java:72
-    public final static String MS_REGISTERING = "Registering";
+    public final static String MS_REGISTERING = "Регистрация";
 //--msgstr "Регистрация"
-    public final static String MS_DONE = "Done";
+    public final static String MS_DONE = "Выполнено";
 //--msgstr "Выполнено"
-    public final static String MS_ERROR_ = "Error: ";
+    public final static String MS_ERROR_ = "Ошибка: ";
 //--msgstr "Ошибка: "
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/ServiceDiscovery.java:42
-    public final static String MS_BROWSE = "Browse";
+    public final static String MS_BROWSE = "Обзор";
 //--msgstr "Обзор"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyModifyList.java:38
-    public final static String MS_SAVE_LIST = "Save list";
+    public final static String MS_SAVE_LIST = "Сохранить список";
 //--msgstr "Сохранить список"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:143
-    public final static String MS_KEEPALIVE_PERIOD = "Keep-Alive period";
+    public final static String MS_KEEPALIVE_PERIOD = "период Keep-Alive";
 //--msgstr "Keep-Alive период"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:105
-    public final static String MS_NEWGROUP = "<New Group>";
+    public final static String MS_NEWGROUP = "<Новая Группа>";
 //--msgstr "<Новая Группа>"
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/DiscoForm.java:39
 // #: /home/avl/Projects/bombus/src/Client/MessageEdit.java:32
-    public final static String MS_SEND = "Send";
+    public final static String MS_SEND = "Отправить";
 //--msgstr "Отправить"
     
 // #: /home/avl/Projects/bombus/src/Client/StatusSelect.java:105
-    public final static String MS_PRIORITY = "Priority";
+    public final static String MS_PRIORITY = "Приоритет";
 //--msgstr "Приоритет"
     
 // #: /home/avl/Projects/bombus/src/ui/SplashScreen.java:96
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:221
-    public final static String MS_FAILED = "Failed";
+    public final static String MS_FAILED = "Ошибка";
 //--msgstr "Ошибка"
     
 // #: /home/avl/Projects/bombus/src/Client/StatusSelect.java:24
-    public final static String MS_SET_PRIORITY = "Set Priority";
+    public final static String MS_SET_PRIORITY = "Установить приоритет";
 //--msgstr "Установить приоритет"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyModifyList.java:34
-    public final static String MS_DELETE_RULE = "Delete rule";
+    public final static String MS_DELETE_RULE = "Удалить правило";
 //--msgstr "Удалить правило"
     
 // // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:94
@@ -265,53 +265,53 @@
 ////--msgstr "Игнор-лист"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:712
-    public final static String MS_ROSTER_REQUEST = "Roster request ";
+    public final static String MS_ROSTER_REQUEST = "Загрузка ростера";
 //--msgstr "Загрузка ростера "
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyForm.java:33
-    public final static String MS_PRIVACY_TYPE = "Type";
+    public final static String MS_PRIVACY_TYPE = "Тип";
 //--msgstr "Тип"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:53
-    public final static String MS_NAME = "Name";
+    public final static String MS_NAME = "Имя";
 //--msgstr "Имя"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:52
-    public final static String MS_USERNAME = "Username";
+    public final static String MS_USERNAME = "Имя пользователя";
 //--msgstr "Имя пользователя"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:130
-    public final static String MS_FULLSCREEN = "fullscreen";
+    public final static String MS_FULLSCREEN = "полноэкранный";
 //--msgstr "полный экран"
     
 // #: /home/avl/Projects/bombus/src/Client/StatusSelect.java:25
-    public final static String MS_ALL_PRIORITIES = "All Priorities";
+    public final static String MS_ALL_PRIORITIES = "для всех статусов";
 //--msgstr "во всех статусах"
     
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:26
-    public final static String MS_ADD_BOOKMARK = "Add bookmark";
+    public final static String MS_ADD_BOOKMARK = "Добав. закладку";
 //--msgstr "Добав. закладку"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:60
-    public final static String MS_CONFERENCES_ONLY = "conferences only";
+    public final static String MS_CONFERENCES_ONLY = "только конференции";
 //--msgstr "только конференции"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:52
-    public final static String MS_CLIENT_INFO = "Client Info";
+    public final static String MS_CLIENT_INFO = "Тип jabber-клиента";
 //--msgstr "Инфо о клиенте"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:80
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:71
-    public final static String MS_DISCARD = "Discard Search";
+    public final static String MS_DISCARD = "Сбросить результаты";
 //--msgstr "Сбросить результаты"
     
-    public final static String MS_SEARCH_RESULTS = "Search Results";
+    public final static String MS_SEARCH_RESULTS = "Результаты поиска";
 //--msgstr "Результаты поиска"
     public final static String MS_GENERAL = "General";
 //--msgstr "General"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:82
-    public final static String MS_MEMBERS = "Members";
+    public final static String MS_MEMBERS = "Члены";
 //--msgstr "Члены"
     
 // // #: /home/avl/Projects/bombus/src/Client/Roster.java:160
@@ -320,68 +320,68 @@
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:49
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:82
-    public final static String MS_ADD_CONTACT = "Add Contact";
+    public final static String MS_ADD_CONTACT = "Новый контакт";
 //--msgstr "Добав. контакт"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:58
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyForm.java:37
-    public final static String MS_SUBSCRIPTION = "Subscription";
+    public final static String MS_SUBSCRIPTION = "Подписка";
 //--msgstr "Подписка"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:78
-    public final static String MS_STATUS_MENU = "Status >";
+    public final static String MS_STATUS_MENU = "Статус >";
 //--msgstr "Статус >"
     
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:24
 // #: /home/avl/Projects/bombus/src/Conference/Bookmarks.java:33
-    public final static String MS_JOIN = "Join";
+    public final static String MS_JOIN = "Присоединиться";
 //--msgstr "Присоединиться"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:117
-    public final static String MS_STARTUP_ACTIONS = "Startup actions";
+    public final static String MS_STARTUP_ACTIONS = "Действия при запуске";
 //--msgstr "Действия при запуске"
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/ServiceDiscovery.java:44
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:54
-    public final static String MS_SERVER = "Server";
+    public final static String MS_SERVER = "Сервер";
 //--msgstr "Сервер"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:81
-    public final static String MS_ADMINS = "Admins";
+    public final static String MS_ADMINS = "Админы";
 //--msgstr "Админы"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacySelect.java:40
-    public final static String MS_MK_ILIST = "Make Ignore-List";
+    public final static String MS_MK_ILIST = "Создать Ignore-List";
 //--msgstr "Создать Игнор-лист"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:89
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:94
-    public final static String MS_OPTIONS = "Options";
+    public final static String MS_OPTIONS = "Настройки";
 //--msgstr "Настройки"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:59
 // #: /home/avl/Projects/bombus/src/Client/AccountSelect.java:34
 // #: /home/avl/Projects/bombus/src/Conference/Bookmarks.java:35
 // #: /home/avl/Projects/bombus/src/archive/ArchiveList.java:31
-    public final static String MS_DELETE = "Delete";
+    public final static String MS_DELETE = "Удалить";
 //--msgstr "Удалить"
-    public final static String MS_DELETE_ASK = "Delete contact?";
+    public final static String MS_DELETE_ASK = "Удалить контакт?";
 //--msgstr "Удалить контакт?"
     
 // #: /home/avl/Projects/bombus/src/Messages/MessageView.java:35
-    public final static String MS_SUBSCRIBE = "Authorize";
+    public final static String MS_SUBSCRIBE = "Авторизовать";
 //--msgstr "Авторизовать"
     
 // #: /home/avl/Projects/bombus/src/Client/MessageEdit.java:34
-    public final static String MS_NICKNAMES = "Nicknames";
+    public final static String MS_NICKNAMES = "Участники";
 //--msgstr "Ники"
     
 // #: /home/avl/Projects/bombus/src/midlet/Bombus.java:90
-    public final static String MS_ENT_SETUP = "Entering setup";
+    public final static String MS_ENT_SETUP = "Вход в настройки аккаунта";
 //--msgstr "Вход в настройки аккаунта"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactMessageList.java:30
-    public final static String MS_ADD_ARCHIVE = "to Archive";
+    public final static String MS_ADD_ARCHIVE = "в Архив";
 //--msgstr "в Архив"
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/ServiceDiscovery.java:46
@@ -398,7 +398,7 @@
 // #: /home/avl/Projects/bombus/src/Messages/MessageList.java:32
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacySelect.java:34
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyModifyList.java:32
-    public final static String MS_BACK = "Back";
+    public final static String MS_BACK = "Назад";
 //--msgstr "Назад"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:132
@@ -406,15 +406,15 @@
 //--msgstr "Монитор кучи"
     
 // #: /home/avl/Projects/bombus/src/ui/SplashScreen.java:115
-    public final static String MS_HIDE_SPLASH = "Hide Splash";
+    public final static String MS_HIDE_SPLASH = "Ростер";
 //--msgstr "Спрятать Заставку"
     
 // #: /home/avl/Projects/bombus/src/Client/StatusSelect.java:112
-    public final static String MS_MESSAGE = "Message";
+    public final static String MS_MESSAGE = "Сообщение";
 //--msgstr "Сообщение"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:79
-    public final static String MS_OTHER = "<Other>";
+    public final static String MS_OTHER = "<Другая>";
 //--msgstr "<Другая>"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:109
@@ -422,35 +422,35 @@
 //--msgstr "история -"
     
 // #: /home/avl/Projects/bombus/src/Conference/AppendNick.java:26
-    public final static String MS_APPEND = "Append";
+    public final static String MS_APPEND = "Добавить";
 //--msgstr "Добавить"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactMessageList.java:33
-    public final static String MS_ACTIVE_CONTACTS = "Active Contacts";
+    public final static String MS_ACTIVE_CONTACTS = "Актив. контакты";
 //--msgstr "Актив. контакты"
     
 // #: /home/avl/Projects/bombus/src/Conference/AppendNick.java:34
-    public final static String MS_SELECT_NICKNAME = "Select nickname";
+    public final static String MS_SELECT_NICKNAME = "Выбрать ник";
 //--msgstr "Выбрать ник"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:54
-    public final static String MS_GROUP = "Group";
+    public final static String MS_GROUP = "Группа";
 //--msgstr "Группа"
     
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:42
-    public final static String MS_JOIN_CONFERENCE = "Join conference";
+    public final static String MS_JOIN_CONFERENCE = "Присоединиться к конференции";
 //--msgstr "Присоединиться к конференции"
     
 // #: /home/avl/Projects/bombus/src/ui/YesNoAlert.java:23
-    public final static String MS_NO = "No";
+    public final static String MS_NO = "Нет";
 //--msgstr "Нет"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:75
-    public final static String MS_REENTER = "Re-Enter Room";
+    public final static String MS_REENTER = "Вернуться в Комнату";
 //--msgstr "Перезайти в Комнату"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactMessageList.java:27
-    public final static String MS_NEW_MESSAGE = "New Message";
+    public final static String MS_NEW_MESSAGE = "Новое сообщение";
 //--msgstr "Новое сообщение"
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/SearchResult.java:28
@@ -459,11 +459,11 @@
 //--msgstr "Добавить"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:45
-    public final static String MS_LOGON = "Logon";
+    public final static String MS_LOGON = "Подключить";
 //--msgstr "Подключить"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:1037
-    public final static String MS_LOGINPGS = "Login in progress";
+    public final static String MS_LOGINPGS = "Соединение...";
 //--msgstr "Соединение..."
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyForm.java:34
@@ -471,23 +471,23 @@
 //--msgstr "Stanzas"
     
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:52
-    public final static String MS_AT_HOST = "at Host";
+    public final static String MS_AT_HOST = "Сервер";
 //--msgstr "Сервер"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:119
-    public final static String MS_AUTO_CONFERENCES = "join conferences";
+    public final static String MS_AUTO_CONFERENCES = "автовход в конференцию";
 //--msgstr "автовход в конференцию"
     
 // #: /home/avl/Projects/bombus/src/Client/StatusSelect.java:32
-    public final static String MS_STATUS = "Status";
+    public final static String MS_STATUS = "Статус";
 //--msgstr "Статус"
     
 // #: /home/avl/Projects/bombus/src/Messages/MessageView.java:33
-    public final static String MS_SMILES_TOGGLE = "Smiles";
+    public final static String MS_SMILES_TOGGLE = "Смайлики";
 //--msgstr "Смайлики"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactMessageList.java:32
-    public final static String MS_CONTACT = "Contact >";
+    public final static String MS_CONTACT = "Контакт >";
 //--msgstr "Контакт >"
     
 // #: /home/avl/Projects/bombus/src/Client/MessageEdit.java:35
@@ -495,15 +495,15 @@
 //--msgstr "/me"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyForm.java:59
-    public final static String MS_ORDER = "Order";
+    public final static String MS_ORDER = "Порядок";
 //--msgstr "Порядок"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:91
-    public final static String MS_OFFLINE_CONTACTS = "offline contacts";
+    public final static String MS_OFFLINE_CONTACTS = "показать отключенных";
 //--msgstr "показать отключенных"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactEdit.java:58
-    public final static String MS_TRANSPORT = "Transport";
+    public final static String MS_TRANSPORT = "Транспорт";
 //--msgstr "Транспорт"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:110
@@ -511,90 +511,90 @@
 //--msgstr "composing события"
     
 // #: /home/avl/Projects/bombus/src/Client/MessageEdit.java:33
-    public final static String MS_ADD_SMILE = "Add Smile";
+    public final static String MS_ADD_SMILE = "Добавить смайлик";
 //--msgstr "Добавить смайлик"
     
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:55
-    public final static String MS_NICKNAME = "Nickname";
+    public final static String MS_NICKNAME = "Ник";
 //--msgstr "Ник"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:65
-    public final static String MS_REVOKE_VOICE = "Revoke Voice";
+    public final static String MS_REVOKE_VOICE = "Отнять право голоса";
 //--msgstr "Отнять право голоса"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:95
-    public final static String MS_NOT_IN_LIST = "Not-in-list";
+    public final static String MS_NOT_IN_LIST = "Не в списке";
 //--msgstr "Не в списке"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:49
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:53
-    public final static String MS_COMMANDS = "Commands";
+    public final static String MS_COMMANDS = "Команды";
 //--msgstr "Команды"
     
 // #: /home/avl/Projects/bombus/src/ui/controls/NumberField.java:39
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:71
-    public final static String MS_CHSIGN = "- (Sign)";
+    public final static String MS_CHSIGN = "- (знак)";
 //--msgstr "- (знак)"
     
 // #: /home/avl/Projects/bombus/src/Client/AlertProfile.java:36
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacySelect.java:36
-    public final static String MS_SETDEFAULT = "Set default";
+    public final static String MS_SETDEFAULT = "Уст. по умолчанию";
 //--msgstr "Уст. по умолчанию"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:83
-    public final static String MS_BANNED = "Outcasts (Ban)";
+    public final static String MS_BANNED = "Изгои (Ban)";
 //--msgstr "Изгои (Ban)"
     
 // #: /home/avl/Projects/bombus/src/Conference/affiliation/AffiliationModify.java:47
-    public final static String MS_SET_AFFILIATION = "Set affiliation to";
+    public final static String MS_SET_AFFILIATION = "Изменить членство на";
 //--msgstr "Изменить членство на"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:88
-    public final static String MS_HIDE_OFFLINES = "Hide Offlines";
+    public final static String MS_HIDE_OFFLINES = "Скрыть отключенных";
 //--msgstr "Скрыть отключенных"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:61
-    public final static String MS_REGISTER_ACCOUNT = "Register Account";
+    public final static String MS_REGISTER_ACCOUNT = "Зарегистрировать аккаунт";
 //--msgstr "Зарегистрировать аккаунт"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:118
-    public final static String MS_AUTOLOGIN = "autologin";
+    public final static String MS_AUTOLOGIN = "автосоединение";
 //--msgstr "автосоединение"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:46
-    public final static String MS_LOGOFF = "Logoff";
+    public final static String MS_LOGOFF = "Отключить";
 //--msgstr "Отключить"
     
 // #: /home/avl/Projects/bombus/src/vcard/vCardForm.java:27
-    public final static String MS_PUBLISH = "Publish";
+    public final static String MS_PUBLISH = "Опубликовать";
 //--msgstr "Опубликовать"
     
 // #: /home/avl/Projects/bombus/src/Client/SubscriptionEdit.java:22
-    public final static String MS_SUBSCR_REMOVE = "Remove subscription";
+    public final static String MS_SUBSCR_REMOVE = "Удалить подписку";
 //--msgstr "Удалить подписку"
     
 // #: /home/avl/Projects/bombus/src/Conference/affiliation/AffiliationModify.java:36
-    public final static String MS_SET = "Set";
+    public final static String MS_SET = "Установить";
 //--msgstr "Установить"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:127
-    public final static String MS_APPLICATION = "Application";
+    public final static String MS_APPLICATION = "Приложение";
 //--msgstr "Приложение"
     
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:25
-    public final static String MS_BOOKMARKS = "Bookmarks";
+    public final static String MS_BOOKMARKS = "Закладки";
 //--msgstr "Закладки"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:72
-    public final static String MS_TEST_SOUND = "Test sound";
+    public final static String MS_TEST_SOUND = "Тест звука";
 //--msgstr "Тест звука"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountRegister.java:42
-    public final static String MS_STARTUP = "Startup";
+    public final static String MS_STARTUP = "Запуск";
 //--msgstr "Запуск"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyModifyList.java:35
-    public final static String MS_EDIT_RULE = "Edit rule";
+    public final static String MS_EDIT_RULE = "Изменить правило";
 //--msgstr "Редакт. правило"
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/ServiceDiscovery.java:47
@@ -614,71 +614,71 @@
 // #: /home/avl/Projects/bombus/src/Conference/ConferenceForm.java:27
 // #: /home/avl/Projects/bombus/src/vcard/vCardForm.java:26
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyForm.java:39
-    public final static String MS_CANCEL = "Cancel";
+    public final static String MS_CANCEL = "Отмена";
 //--msgstr "Отмена"
     
 // #: /home/avl/Projects/bombus/src/Info/InfoWindow.java:29
-    public final static String MS_CLOSE = "Close";
+    public final static String MS_CLOSE = "Закрыть";
 //--msgstr "Закрыть"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:89
 // #: /home/avl/Projects/bombus/src/Client/MessageEdit.java:37
 // #: /home/avl/Projects/bombus/src/archive/ArchiveList.java:54
-    public final static String MS_ARCHIVE = "Archive";
+    public final static String MS_ARCHIVE = "Архив";
 //--msgstr "Архив"
-    public final static String MS_FREE = "free ";
+    public final static String MS_FREE = "свободно ";
 //--msgstr "свободно "
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:86
-    public final static String MS_CONFERENCE = "Conference";
+    public final static String MS_CONFERENCE = "Конференция";
 //--msgstr "Конференция"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:148
-    public final static String MS_SOUND = "Sound";
+    public final static String MS_SOUND = "Звук";
 //--msgstr "Звук"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:667
-    public final static String MS_LOGIN_FAILED = "Login failed";
+    public final static String MS_LOGIN_FAILED = "Ошибка авторизации";
 //--msgstr "Ошибка авторизации"
     
 // #: /home/avl/Projects/bombus/src/ServiceDiscovery/ServerBox.java:25
-    public final static String MS_DISCOVER = "Browswe"; //"Discover"
+    public final static String MS_DISCOVER = "Открыть"; //"Discover"
 //--msgstr "Открыть"
     
 // #: /home/avl/Projects/bombus/src/Conference/affiliation/Affiliations.java:44
-    public final static String MS_NEW_JID = "New Jid";
+    public final static String MS_NEW_JID = "Новый Jid";
 //--msgstr "Новый Jid"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:59
-    public final static String MS_PLAIN_PWD = "plain-text password";
+    public final static String MS_PLAIN_PWD = "пароль откр. текстом";
 //--msgstr "пароль откр. текстом"
     
 // #: /home/avl/Projects/bombus/src/archive/ArchiveList.java:35
-    public final static String MS_PASTE_NICKNAME = "Paste Nickname";
+    public final static String MS_PASTE_NICKNAME = "Вставить ник";
 //--msgstr "Вставить ник"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:62
-    public final static String MS_KICK = "Kick";
+    public final static String MS_KICK = "Выгнать (kick)";
 //--msgstr "Выгнать (kick)"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactMessageList.java:31
-    public final static String MS_CLEAR_LIST = "Clear List";
+    public final static String MS_CLEAR_LIST = "Очистить список";
 //--msgstr "Очистить список"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:64
-    public final static String MS_GRANT_VOICE = "Grant Voice";
+    public final static String MS_GRANT_VOICE = "Дать право голоса";
 //--msgstr "Дать право голоса"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyModifyList.java:37
     public final static String MS_MOVE_DOWN = "Move Down";
-//--msgstr "Переместить вниз"
+//--msgstr "Сместить вниз"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactMessageList.java:29
-    public final static String MS_QUOTE = "Quote";
+    public final static String MS_QUOTE = "Цитата";
 //--msgstr "Цитата"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:90
-    public final static String MS_ROSTER_ELEMENTS = "Roster elements";
+    public final static String MS_ROSTER_ELEMENTS = "Элементы ростера";
 //--msgstr "Элементы ростера"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:136
@@ -686,46 +686,46 @@
 //--msgstr "Всплывать"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:108
-    public final static String MS_SMILES = "smiles";
+    public final static String MS_SMILES = "смайлики";
 //--msgstr "смайлики"
     
 // #: /home/avl/Projects/bombus/src/Info/InfoWindow.java:28
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:95
-    public final static String MS_ABOUT = "About";
+    public final static String MS_ABOUT = "О программе";
 //--msgstr "О программе"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:66
-    public final static String MS_RESOURCE = "Resource";
+    public final static String MS_RESOURCE = "Ресурс";
 //--msgstr "Ресурс"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:1068
-    public final static String MS_DISCONNECTED = "Disconnected";
+    public final static String MS_DISCONNECTED = "Соединение разорвано";
 //--msgstr "Соединение разорвано"
     
 // #: /home/avl/Projects/bombus/src/Client/StatusSelect.java:23
 // #: /home/avl/Projects/bombus/src/Client/RosterItemActions.java:57
 // #: /home/avl/Projects/bombus/src/Client/AccountSelect.java:33
-    public final static String MS_EDIT = "Edit";
+    public final static String MS_EDIT = "Редактировать";
 //--msgstr "Редактировать"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:55
-    public final static String MS_HOST_IP = "Host name/IP (optional)";
+    public final static String MS_HOST_IP = "Имя хоста/IP (необязательно)";
 //--msgstr "Имя хоста/IP (необязательно)"
     
 // #: /home/avl/Projects/bombus/src/PrivacyLists/PrivacyModifyList.java:33
-    public final static String MS_ADD_RULE = "Add rule";
+    public final static String MS_ADD_RULE = "Добавить правило";
 //--msgstr "Добавить правило"
     
 // #: /home/avl/Projects/bombus/src/Client/StatusSelect.java:109
-    public final static String MS_ALL_STATUSES = "for all status types";
+    public final static String MS_ALL_STATUSES = "для всех статсуов";
 //--msgstr "для статсуов всех типов"
     
 // #: /home/avl/Projects/bombus/src/archive/ArchiveList.java:34
-    public final static String MS_PASTE_JID = "Paste Jid";
+    public final static String MS_PASTE_JID = "Вставить Jid";
 //--msgstr "Вставить Jid"
     
 // #: /home/avl/Projects/bombus/src/Messages/MessageView.java:37
-    public final static String MS_GOTO_URL = "Goto URL";
+    public final static String MS_GOTO_URL = "Перейти к URL";
 //--msgstr "Перейти к URL"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:96
@@ -734,39 +734,39 @@
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:218
 // #: /home/avl/Projects/bombus/src/Client/AccountSelect.java:30
-    public final static String MS_LOGIN = "Login";
+    public final static String MS_LOGIN = "Соединиться";
 //--msgstr "Соединиться"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:145
-    public final static String MS_CLOCK_OFFSET = "Clock offset";
+    public final static String MS_CLOCK_OFFSET = "Смещение часов";
 //--msgstr "Смещение часов"
     
 // #: /home/avl/Projects/bombus/src/ui/YesNoAlert.java:22
-    public final static String MS_YES = "Yes";
+    public final static String MS_YES = "Да";
 //--msgstr "Да"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:134
-    public final static String MS_FLASHBACKLIGHT = "flash backlight";
+    public final static String MS_FLASHBACKLIGHT = "Мигать подсветкой";
 //--msgstr "Мигать подсветкой"
     
 // #: /home/avl/Projects/bombus/src/Client/MessageEdit.java:30
-    public final static String MS_SUSPEND = "Suspend";
+    public final static String MS_SUSPEND = "Отложить";
 //--msgstr "Отложить"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:84
-    public final static String MS_ALERT_PROFILE_CMD = "Alert Profile >";
-//--msgstr "Сигнализация >"
+    public final static String MS_ALERT_PROFILE_CMD = "Сигналы >";
+//--msgstr "Сигналы >"
     
 // #: /home/avl/Projects/bombus/src/Client/RosterToolsMenu.java:32
-    public final static String MS_MY_VCARD = "My vCard";
+    public final static String MS_MY_VCARD = "Личная vCard";
 //--msgstr "Личная vCard"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:93
-    public final static String MS_TRANSPORTS = "transports";
+    public final static String MS_TRANSPORTS = "транспорты";
 //--msgstr "транспорты"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountSelect.java:32
-    public final static String MS_NEW_ACCOUNT = "New Account";
+    public final static String MS_NEW_ACCOUNT = "Новый аккаунт";
 //--msgstr "Новый аккаунт"
     
 // #: /home/avl/Projects/bombus/src/Client/ConfigForm.java:92
@@ -779,22 +779,23 @@
 //--msgstr "vCard"
     
 // #: /home/avl/Projects/bombus/src/Client/MessageEdit.java:36
-    public final static String MS_SET_SUBJECT = "Set Subject";
+    public final static String MS_SET_SUBJECT = "Установить тему";
 //--msgstr "Установить тему"
     
 // #: /home/avl/Projects/bombus/src/Client/Roster.java:91
-    public final static String MS_TOOLS = "Tools";
+    public final static String MS_TOOLS = "Инструменты";
 //--msgstr "Инструменты"
     
-    public final static String MS_JABBER_TOOLS = "Jabber Tools";
+    public final static String MS_JABBER_TOOLS = "Инструменты сервера";
 //--msgstr "Инструменты сервера"
     
 // #: /home/avl/Projects/bombus/src/Client/AccountForm.java:56
-    public final static String MS_PORT = "Port";
+    public final static String MS_PORT = "Порт";
 //--msgstr "Порт"
     
 // #: /home/avl/Projects/bombus/src/Client/ContactMessageList.java:28
-    public final static String MS_RESUME = "Resume Message";
+    public final static String MS_RESUME = "Черновик";
+//--msgstr "Продолж. сообщ."
     
     public final static String MS_PROXY_ENABLE = "proxy CONNECT";
     
@@ -809,7 +810,6 @@
     public final static String MS_MODIFY_AFFILIATION = "Modify affiliation";
 
     public final static String MS_AUTOFOCUS = "Autofocus";
-//--msgstr "Продолж. сообщ."
     
     /** Creates a new instance of SR */
     private SR() { }
