Index: resources/text/lang/ru.txt
===================================================================
--- resources/text/lang/ru.txt	(revision 529)
+++ resources/text/lang/ru.txt	(working copy)
@@ -224,4 +224,13 @@
 //Grant Ownership	Дать право владельца
 //Revoke Ownership	Отнять право владельца
 
-//room presences
+room presences	Показ. ст. конф.
+
+ is now known as 	 сменил ник на 
+ was banned 	 запретили входить в конату 
+ was kicked 	 выгнали из комнаты 
+ has been kicked because room became members-only	 выгнан, т.к. комната теперь только для членов
+ has left the channel	 вышел из конференции
+ has joined the channel as 	 подключился как 
+ and 	 и 
+ is now 	 сменил статус на 
\ No newline at end of file
Index: src/Conference/MucContact.java
===================================================================
--- src/Conference/MucContact.java	(revision 529)
+++ src/Conference/MucContact.java	(working copy)
@@ -13,6 +13,7 @@
 import Client.StaticData;
 import com.alsutton.jabber.JabberDataBlock;
 import com.alsutton.jabber.datablocks.Presence;
+import locale.SR;
 
 /**
  *
@@ -95,7 +96,7 @@
             switch (statusCode) {
                 
                 case 303:
-                    b.append(" is now known as ");
+                    b.append(SR.MS_IS_NOW_KNOWN_AS);
                     b.append(chNick);
                     // исправим jid
                     String newJid=from.substring(0,rp+1)+chNick;
@@ -109,7 +110,7 @@
                 case 307: //kick
                 case 301: //ban
                     b.append(
-                            (statusCode==301)? " was banned " : " was kicked " );
+                            (statusCode==301)? SR.MS_WAS_BANNED : SR.MS_WAS_KICKED );
                     b.append("(");
                     b.append(reason);
                     b.append(")");
@@ -117,12 +118,12 @@
                     break;
             
                 case 322:
-                    b.append(" has been kicked because room became members-only");
+                    b.append(SR.MS_HAS_BEEN_KICKED_BECAUSE_ROOM_BECAME_MEMBERS_ONLY);
                     testMeKicked();
                     break;
                     
                 default:
-                b.append(" has left the channel");
+                b.append(SR.MS_HAS_LEFT_CHANNEL);
             } 
                 
         } else {
@@ -134,10 +135,10 @@
                     b.append(')');
                     this.realJid=realJid;  //for moderating purposes
                 }
-                b.append(" has joined the channel as ");
+                b.append(SR.MS_HAS_JOINED_THE_CHANNEL_AS);
                 b.append(role);
                 if (!affiliation.equals("none")) {
-                    b.append(" and ");
+                    b.append(SR.MS_AND);
                     b.append(affiliation);
 //toon
                     //b.append(" with status ");
@@ -145,7 +146,7 @@
                     
                 }
             } else {
-                b.append(" is now ");
+                b.append(SR.MS_IS_NOW);
                 if ( roleChanged ) b.append(role);
                 if (affiliationChanged) {
                     if (roleChanged) b.append(" and ");
Index: src/locale/SR.java
===================================================================
--- src/locale/SR.java	(revision 529)
+++ src/locale/SR.java	(working copy)
@@ -232,6 +232,15 @@
     
     public   static String MS_STORE_PRESENCE = loadString( "room presences" );
     
+    public static String MS_IS_NOW_KNOWN_AS=loadString(" is now known as ");
+    public static String MS_WAS_BANNED=loadString(" was banned ");
+    public static String MS_WAS_KICKED=loadString(" was kicked ");
+    public static String MS_HAS_BEEN_KICKED_BECAUSE_ROOM_BECAME_MEMBERS_ONLY=loadString(" has been kicked because room became members-only");
+    public static String MS_HAS_LEFT_CHANNEL=loadString(" has left the channel");
+    public static String MS_HAS_JOINED_THE_CHANNEL_AS=loadString(" has joined the channel as ");
+    public static String MS_AND=loadString(" and ");
+    public static String MS_IS_NOW=loadString(" is now ");    
+    
 
     public static String MS_XMLLANG;
     public static String MS_IFACELANG;
