Index: ui/VirtualList.java
===================================================================
--- ui/VirtualList.java	(revision 505)
+++ ui/VirtualList.java	(working copy)
@@ -87,7 +87,7 @@
     public static final int MOTOROLA_GREEN=-10;
     public final static int MOTOROLA_FLIP=-200;
     public static final int SE_GREEN=0;
-    
+    public static final int SE_CLEAR=-8;
     public static int greenKeyCode=SIEMENS_GREEN;
     public static boolean fullscreen=false;
     public static boolean memMonitor;
Index: Client/ContactMessageList.java
===================================================================
--- Client/ContactMessageList.java	(revision 505)
+++ Client/ContactMessageList.java	(working copy)
@@ -145,4 +145,13 @@
 	if (keyCode==KEY_NUM3) new ActiveContacts(display, contact);
 	else super.keyPressed(keyCode);
     }
+	public void keyPressed(int keyCode) {
+            if (keyCode==SE_CLEAR) {
+            contact.purge();
+            messages=new Vector();
+            System.gc();
+            redraw();
+            }
+             else super.keyPressed(keyCode);
+    }
 }
