Index: src/Info/Version.java
===================================================================
--- src/Info/Version.java	(revision 620)
+++ src/Info/Version.java	(working copy)
@@ -22,7 +22,8 @@
     
     public final static String url="http://bombus.jrudevels.org";
 
-
+    public static boolean isSiemens;
+	
     private static String platformName;
     
     public static String getPlatformName() {
@@ -55,6 +56,13 @@
                 if (device==null) device=System.getProperty("funlights.product");
                 if (device!=null) platformName="Motorola-"+device;
             }
+            if (platformName.indexOf("SIE") > -1) {
+                platformName=System.getProperty("microedition.platform");
+                isSiemens=true;
+            } else if (System.getProperty("com.siemens.OSVersion")!=null) {
+                platformName="SIE-"+System.getProperty("microedition.platform")+"/"+System.getProperty("com.siemens.OSVersion");
+                isSiemens=true;
+            }
         }
         return platformName;
     }
